docker/deployer/Dockerfile

```diff
@@ -37,5 +35,43 @@ RUN poetry install
 COPY . .
 RUN make setup && make build
 
+FROM python:3.9.13-slim-bullseye
+
+WORKDIR /app/kakarot
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl \
+    apt-transport-https \
+    ca-certificates \
+    gcc \
+    libgmp-dev \
+    python3-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -sSL https://install.python-poetry.org | python3 -
+
+ENV PATH="$PATH:/root/.local/bin"
+ENV PYTHONPATH="/app/kakarot"
+RUN poetry config virtualenvs.create false
+
+# Copy only necessary artifacts from the builder stage
+# Correctly copy the deploy_kakarot.py script to its intended location
+COPY --from=builder /build/scripts/ /app/kakarot/scripts/
+COPY --from=builder /build/build /app/kakarot/build/
+COPY --from=builder /build/Makefile /app/kakarot/
+COPY --from=builder /build/tests /app/kakarot/tests/
+
+# Cairo Smart contracts are used in deploy_kakarot.py
+# To limit the probability of this Dockerfile to break, we copy the entire src and not individual files
+COPY --from=builder /build/src /app/kakarot/src/
+
+# Copy poetry lock and toml
+COPY --from=builder /build/poetry.lock /app/kakarot/
+COPY --from=builder /build/pyproject.toml /app/kakarot/
+
+# Copy foundry.toml
+COPY --from=builder /build/foundry.toml /app/kakarot/
+
+RUN poetry install
```

Comment:
do the install the earliest possible (before copying the project) to leverage docker image cache